gobblin-restli/gobblin-job-config-service/server/src/main/java/gobblin/service/JobConfigsResource.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobblin.service;
+
+import com.google.common.collect.Maps;
+import com.linkedin.restli.common.ComplexResourceKey;
+import com.linkedin.restli.common.EmptyRecord;
+import com.linkedin.restli.common.HttpStatus;
+import com.linkedin.restli.server.RestLiServiceException;
+import com.linkedin.restli.server.UpdateResponse;
+import com.linkedin.restli.server.annotations.RestLiCollection;
+import com.linkedin.restli.server.CreateResponse;
+import com.linkedin.data.template.StringMap;
+import com.linkedin.restli.server.resources.ComplexKeyResourceTemplate;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import gobblin.config.ConfigBuilder;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.runtime.api.JobSpec;
+import gobblin.runtime.api.JobSpecNotFoundException;
+import gobblin.runtime.api.MutableJobCatalog;
+import gobblin.runtime.job_catalog.FSJobCatalog;
+import gobblin.runtime.job_catalog.ImmutableFSJobCatalog;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@RestLiCollection(name = "jobconfigs", namespace = "gobblin.service")
+/**
+ * Rest.li resource for handling job configuration requests
+ */
+public class JobConfigsResource extends ComplexKeyResourceTemplate<JobConfigId, EmptyRecord, JobConfig> {
+  private static final Logger LOG = LoggerFactory.getLogger(JobConfigsResource.class);
+
+  private MutableJobCatalog jobCatalog;
```

Comment:
You should add an `@Inject` annotation. That's the way rest.li resources are supposed to work.